### PR TITLE
(Archiving) Refactor posts

### PR DESCRIPTION
**Pull request by islean**
## Description

This PR refactors where requests are made towards Miria/DDN. The main difference is that the Request/Payload/DTO's don't carry that functionality.

### Changed

- Posts are not part of the Payload classes.

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b archiving-refactor-requests -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/meta/archive/ddn/models.py [hunk 1]
```diff
@@ -72,40 +69,16 @@ def add_repositories(self, source_prefix: str, destination_prefix: str):
...
+    """Model representing the response fields of an archive request to the Dataflow
+    API."""
+
+    job_id: int = Field(alias="jobId")
 
-class TransferResponse(BaseModel):
-    """Model representing th response fields of an archive or retrieve reqeust to the Dataflow
+
+class RetrievalResponse(BaseModel):
+    """Model representing the response fields of a retrieval reqeust to the Dataflow
```
Comment (ChrOertlin) [inline cg/meta/archive/ddn/models.py:81]: Is it for better typehinting to have two models holding the same information, i.e. a job_id? 

Can't they both be ```ResponseId```?

File: cg/meta/archive/ddn/ddn_data_flow_client.py [hunk 2]
```diff
@@ -202,7 +209,54 @@ def delete_file(self, file_and_sample: FileAndSample) -> None:
...
             headers=dict(self.headers, **self.auth_header),
+            body=delete_file_payload,
+        )
+        delete_file_response = DeleteFileResponse.model_validate(response.json())
+        if delete_file_response.message != DELETE_FILE_SUCCESSFUL_MESSAGE:
+            raise DdnDataflowDeleteFileError(
+                f"Deletion failed with message {delete_file_response.message}"
+            )
+
+    def _archive_file(self, body: BaseModel, headers: dict) -> ArchivalResponse:
```
Comment (diitaz93) [inline cg/meta/archive/ddn/ddn_data_flow_client.py:223]: A docstring would be nice here

**Review by ChrOertlin — APPROVED** [review]
Looks good 👍

File: cg/meta/archive/ddn/ddn_data_flow_client.py [hunk 3]
```diff
@@ -178,11 +186,10 @@ def is_job_done(self, job_id: int) -> bool:
         """Returns True if the specified job is completed, and false if it is still ongoing.
         Raises:
             ArchiveJobFailedError if the specified job has a failed status."""
-        get_job_status_payload = GetJobStatusPayload(id=job_id)
-        get_job_status_response: GetJobStatusResponse = get_job_status_payload.get_job_status(
-            url=urljoin(self.url, DataflowEndpoints.GET_JOB_STATUS + str(job_id)),
-            headers=dict(self.headers, **self.auth_header),
+        get_job_status_response: GetJobStatusResponse = self._get_job_status(
```
[the remark this answers is not in the document]
Comment (islean) [reply cg/meta/archive/ddn/ddn_data_flow_client.py:189]: I don't mind either, as long as we are consistent in what we call the endpoint and the action of fetching data from it. For me, I would expect a "GetJobStatusResponse" to be a DTO carrying the resulting data from a "GetJobStatus" call. If we call it "JobStatusResponse", I would expect it to be a DTO carrying the resulting data from a "JobStatus" call. 

Given that the rest of the requests/responses are named after the action, e.g. "ArchiveFileResponse", this feels the most consistent to me.

File: cg/meta/archive/ddn/ddn_data_flow_client.py [hunk 4]
```diff
@@ -202,7 +209,54 @@ def delete_file(self, file_and_sample: FileAndSample) -> None:
...
         )
-        delete_file_payload.delete_file(
-            url=urljoin(self.url, DataflowEndpoints.DELETE_FILE),
+        response: Response = self._post_request(
+            endpoint=DataflowEndpoints.DELETE_FILE,
             headers=dict(self.headers, **self.auth_header),
+            body=delete_file_payload,
+        )
+        delete_file_response = DeleteFileResponse.model_validate(response.json())
+        if delete_file_response.message != DELETE_FILE_SUCCESSFUL_MESSAGE:
```
[the remark this answers is not in the document]
Comment (islean) [reply cg/meta/archive/ddn/ddn_data_flow_client.py:218]: The DeleteFileResponse is determined by the Miria/DDN API, so we can not do that. We could perhaps add it as a property, but I would rather keep the class as a clean DTO.

File: cg/meta/archive/ddn/models.py (hunk 1 above)
Comment (islean) [reply cg/meta/archive/ddn/models.py:81]: I felt that this was clearer for me to read. Now we see what archival request will return and what the retrieval request will return. Also, ArchivalResponse and RetrievalResponse classes contain only a subset of the returned values from the endpoint. So we could potentially in the future have a need for them to have different fields as well, should we have a need to parse more of them from the responses.

File: cg/meta/archive/ddn/ddn_data_flow_client.py (hunk 2 above)
Comment (islean) [reply cg/meta/archive/ddn/ddn_data_flow_client.py:223]: Added!

File: cg/meta/archive/ddn/ddn_data_flow_client.py [hunk 5]
```diff
@@ -202,7 +209,58 @@ def delete_file(self, file_and_sample: FileAndSample) -> None:
         delete_file_payload = DeleteFilePayload(
             global_path=f"{self.archive_repository}{sample_id}/{file_name}"
         )
-        delete_file_payload.delete_file(
-            url=urljoin(self.url, DataflowEndpoints.DELETE_FILE),
+        response: Response = self._post_request(
+            endpoint=DataflowEndpoints.DELETE_FILE,
             headers=dict(self.headers, **self.auth_header),
```
Comment (islean) [inline cg/meta/archive/ddn/ddn_data_flow_client.py:214]: Would we want this dict as a class variable? e.g. self.authorized_headers or something?

Comment (diitaz93) [reply cg/meta/archive/ddn/ddn_data_flow_client.py:214]: is it used in many other places?

**Review by diitaz93 — APPROVED** [review]
Looks good 👍 I can't spot any obvious mistake and the implementation is quite clear

File: cg/meta/archive/ddn/ddn_data_flow_client.py (hunk 5 above)
Comment (islean) [reply cg/meta/archive/ddn/ddn_data_flow_client.py:214]: It is used in all posts except for when we fetch authorization tokens in the beginning.

**Review by Vince-janv — APPROVED** [review]
Nice! Love the use of alias 🌟

File: cg/meta/archive/ddn/ddn_data_flow_client.py (hunk 5 above)
Comment (islean) [reply cg/meta/archive/ddn/ddn_data_flow_client.py:214]: In another PR if that's the case

**Comment by islean** [comment]
Deployed to production:

```
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/isak.ohlsson
done.
Log deploy... done.
cg, version 55.4.8
θ63° [isak.ohlsson@hasta:~] [P_base] 3m9s $ 
```

**Pull request merged**